runtime/jvmti/jvmtiStartup.c

```diff
@@ -1110,6 +1282,77 @@ isAgentLibraryLoaded(J9JavaVM *vm, const char *library, BOOLEAN decorate)
 			j9mem_free_memory(agentPath);
 		}
 		Trc_JVMTI_isAgentLibraryLoaded_result(library, agentPath, result);
+#elif defined(AIXPPC) /* defined(LINUX) || defined(OSX) || defined(WIN32) */
+		/* The incoming library & decorate could be in the following cases:
+		 * - JvmtiAgent1 & TRUE
+		 * - libJvmtiAgent1.so & FALSE
+		 * - /path/to/libJvmtiAgent1.so & FALSE
+		 */
+		struct ld_info *objectFiles = NULL;
+		UDATA objectFileSize = 150 * sizeof(struct ld_info);
+		BOOLEAN success = FALSE;
+		PORT_ACCESS_FROM_JAVAVM(vm);
+
+		/* retrieve a list of all object files loaded for the current process */
+		success = loadqueryHelper(vm, L_GETINFO, (void **)&objectFiles, &objectFileSize);
+		Trc_JVMTI_isAgentLibraryLoaded_loadquery_getinfo(objectFiles, objectFileSize, success);
+		if (success) {
+			char *libNameSo = NULL;
+			char *libNameAr = NULL;
+			char *libpath = NULL;
+			UDATA libpathSize = 1024;
+			char *fileName = NULL;
+			BOOLEAN isLibraryPath = FALSE;
+			struct ld_info *cursor = NULL;
+
+			if (decorate) {
+				/* the library is a platform independent name */
+				UDATA libraryLen = strlen(library);
+				/* either object suffix is acceptable */
+				UDATA sizeLibNameSo = libraryLen + sizeof("lib.so");
+				UDATA sizeLibNameAr = libraryLen + sizeof("lib.a");
+				libNameSo = (char *)j9mem_allocate_memory(sizeLibNameSo, J9MEM_CATEGORY_JVMTI);
+				libNameAr = (char *)j9mem_allocate_memory(sizeLibNameAr, J9MEM_CATEGORY_JVMTI);
+				if ((NULL != libNameSo) && (NULL != libNameAr)) {
+					j9str_printf(PORTLIB, libNameSo, sizeLibNameSo, "lib%s.so", library);
+					j9str_printf(PORTLIB, libNameAr, sizeLibNameAr, "lib%s.a", library);
+					Trc_JVMTI_isAgentLibraryLoaded_libNames(libNameSo, libNameAr);
+				} else {
+					Trc_JVMTI_isAgentLibraryLoaded_OOM(libNameSo, libNameAr);
+					vm->internalVMFunctions->setNativeOutOfMemoryError(vm->mainThread, 0, 0);
+					goto failed;
+				}
+			} else {
+				/* check if library contains a path separator */
+				isLibraryPath = (NULL != strrchr(library, '/'));
+			}
+
+			/* retrieve the library path */
+			loadqueryHelper(vm, L_GETLIBPATH, (void **)&libpath, &libpathSize);
+			Trc_JVMTI_isAgentLibraryLoaded_loadquery_getlibpath(libpath, libpathSize);
+			/* regardless of the libpath result, search the object files to check if the library was loaded */
```

Comment:
Ignoring the result may yield `libpath == NULL` will causes problems later, e.g. in call to `strtok_r()`.

Reply:
Assigned `libpath` an empty string when it returns `NULL`.